source/connection-troubleshooting.txt

```diff
@@ -0,0 +1,325 @@
+verify the correct port is open in your firewall.
+
+.. warning::
+
+   Do not open a port in your firewall unless you are sure that it's the port
+   used by your MongoDB instance.
+
+ECONNREFUSED Error
+~~~~~~~~~~~~~~~~~~
+
+If the connection is refused when the driver attempts to connect to the MongoDB
+instance, it generates this error message:
+
+.. code-block:: none
+   :copyable: false
+
+   MongoServerSelectionError: connect ECONNREFUSED ::1:27017
+
+The following sections describe methods that may help resolve the issue.
+
+Start mongod in IPV6 Mode
+-------------------------
+
+Beginning with version 17, Node.js defaults to using ``IPV6``. You can start
+your ``mongod`` in ``IPV6`` mode with
+:manual:`the --ipv6 flag </reference/program/mongod/#std-option-mongod.--ipv6>`,
+or setting the ``net.ipv6`` option to ``true``
+:manual:`in your configuration file </reference/configuration-options/#mongodb-setting-net.ipv6>`.
+
+.. code-block:: sh
+   :caption: Starting mongod with the --ipv6 flag
+
+   mongod --ipv6
+
+.. code-block:: yaml
+   :caption: Setting the net.ipv6 option in the configuration file
+
+   net:
+     ipv6: true
+
+Use IPV4 with Your Mongo Client
+-------------------------------
+
+Beginning with version 17, Node.js defaults to using ``IPV6``. You can
+explicitly use ``IPV4`` with your client by specifying ``family: 4`` as an
+`option to your MongoClient <{+api+}/interfaces/MongoClientOptions.html#family>`__.
+
+.. code-block:: js
+
+   const client = new MongoClient(uri, {
+     family: 4,
+   });
+
+ECONNRESET Error
+~~~~~~~~~~~~~~~~
+
+If the connection is reset when the driver calls ``client.connect()``, it
+generates this error message:
+
+.. code-block:: none
+   :copyable: false
+
+   MongoServerSelectionError: connect ECONNRESET ::1:27017
+
+The following section describes a method that may help resolve the issue.
+
+Increase Number of File Descriptors
+-----------------------------------
+
+In most operating systems, each connection is associated with a `file
+descriptor
+<https://www.computerhope.com/jargon/f/file-descriptor.htm>`_. There is
+typically a limit set by the operating system on the number of file
+descriptors used by a single process. An ``ECONNRESET`` error can occur
+if the connection pool size surpasses the limit of ``file descriptors``.
+
+Consider the following operation:
+
+.. literalinclude:: /code-snippets/faq/econnresetWithClientConnect-example.js
+  :language: javascript
+  :linenos:
+
+If this operation causes an ``ECONNRESET`` error, you may have run into
+the ``file descriptor`` limit for your Node.js process. In that case you
+must increase the number of ``file descriptors`` for the Node.js process. On
+MacOS and Linux you do this with the `ulimit
+<https://ss64.com/bash/ulimit.html>`_ shell command.
+
+.. code-block:: sh
+
+   ulimit -n 6000
+
+This sets the maximum number of ``file descriptors`` for the process to
+6000, allowing Node.js to connect with a pool size of 5000 sockets.
```

Comment:
Suggestion:
I think file descriptors are used for more than just connections, so maybe there is no guarantee. I think it is sufficient to draw the connection between file descriptors and connections, perhaps earlier in the description.

E.g.
"The driver needs to use file descriptors to reference each connection in its connection pool."